src/output-filestore.c

```diff
@@ -492,9 +493,14 @@ static OutputInitResult OutputFilestoreLogInitCtx(ConfNode *conf)
                        "from conf file - %s.  Killing engine",
                        stream_depth_str);
             exit(EXIT_FAILURE);
-        } else {
-            FileReassemblyDepthEnable(stream_depth);
         }
+        if (stream_depth && stream_depth <= stream_config.reassembly_depth) {
+            SCLogWarning(SC_WARN_FILESTORE_CONFIG,
+                       "file-store.stream-depth value %" PRIu32 " has "
+                       "no effect since it's less than stream.reassembly.depth "
+                       "value.", stream_depth);
+        } 
+        FileReassemblyDepthEnable(stream_depth);
```

Comment:
the message above says that `stream_depth` has no effect, yet it is used here regardless. Is that an error?

Reply:
This is intentional but we can change the behavior. Since the configured value is less than `stream.reassembly.depth` its effect is NIL.

Test case `filestore-v2.8-stream-depth` covers this.

Reply:
I think its clearer to not call this function if the value is determined to have no affect. I have to jump through a few functions to verify this in code.

Reply:
Agree with Jason.